### PR TITLE
docs: added add-ons

### PR DESCRIPTION
**Pull request by connor-leech**
We've had some questions regarding our add-on strategy. Although the add-on model remains in flux, we're adding the page back to the site with pared-down copy.

### CODE REVIEW

File: src/content/contributions/add-ons/add-ons.mdx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+---
+title: Add-ons
+---
+
+import { CheckmarkFilled16 } from '@carbon/icons-react';
+
+<AnchorLinks>
+
+- [Add-on repositories](#add-on-repositories)
```
Comment (joshblack) [inline src/content/contributions/add-ons/add-ons.mdx]: In the past we've referred to them as "Carbon Add-ons" versus including repos in the name, is referring to them as "add-on repos" intentional?

File: src/content/contributions/add-ons/add-ons.mdx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+- [Contributing to add-ons](#contributing-to-add-ons)
+- [Ownership and maintenance](#ownership-and-maintenance)
+
+</AnchorLinks>
+
+## Add-on repositories
+
+Add-on repositories, or repos, contain components built for a specific product or experience. The repos are built on top of the core Carbon repo, and enable teams to create their own custom components that follow Carbon's visual style and guidelines.
```
Comment (joshblack) [inline src/content/contributions/add-ons/add-ons.mdx]: > The repos are built on top of the core Carbon repo, and enable teams to create

This sentence seems to imply a hierarchy that may not necessarily be there. Add-on components should be built following the Carbon Design System visual style/guideilnes (namely tokens) but don't necessarily have any connection to the core Carbon repo.

File: src/content/contributions/add-ons/add-ons.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+If your team is using Carbon and needs components specific to your product or industry, host the custom components in an add-on repo. We have created a [React boilerplate](https://github.com/carbon-design-system/carbon-addons-boilerplate-react) and a [Vanilla boilerplate](https://github.com/carbon-design-system/carbon-boilerplate) that you can clone to get started.
```
Comment (joshblack) [inline src/content/contributions/add-ons/add-ons.mdx]: This usage of repo seems perfect, however, do we want to call out the boilerplates? Their super outdated and I would not recommend them personally.

Comment (connor-leech) [reply src/content/contributions/add-ons/add-ons.mdx]: If you don't recommend calling out the boilerplates we most definitely won't.

File: src/content/contributions/add-ons/add-ons.mdx (hunk 2 above)
Comment (connor-leech) [reply src/content/contributions/add-ons/add-ons.mdx]: 👍

File: src/content/contributions/add-ons/add-ons.mdx (hunk 1 above)
Comment (connor-leech) [reply src/content/contributions/add-ons/add-ons.mdx]: I was under the impression add-ons were explicitly repositories. Will adjust.

**Pull request merged**